src/Microsoft.PowerShell.IoT/Microsoft.PowerShell.IoT.cs

```
@@ -247,47 +287,57 @@ public class GetGpioPin : Cmdlet
+                    try
+                    {
+                        pin.PinMode = Unosquare.RaspberryIO.Gpio.GpioPinDriveMode.Input;
+                        if (this.PullMode.HasValue)
+                        {
+                            pin.InputPullMode = (Unosquare.RaspberryIO.Gpio.GpioPinResistorPullMode)this.PullMode.Value;
+                        };
+                    }
+                    catch (System.NotSupportedException)
+                    {
+                        // We want to avoid errors like
+                        // System.NotSupportedException : Get - GpioPin : Pin Pin15 'BCM 14 (UART Transmit)' does not support mode 'Input'.Pin capabilities are limited to: UARTTXD
+                        // at the same time we need to return PinInfo for such pins, so we need to continue processing
+                    }
+                    bool pinBoolValue = pin.Read();
+
+                    if (this.Raw)
+                    {
+                        WriteObject(pinBoolValue ? SignalLevel.High : SignalLevel.Low);
+                    }
+                    else
+                    {
+                        GpioPinData pinData = new GpioPinData(pinId, pinBoolValue ? SignalLevel.High : SignalLevel.Low, pin);
+                        WriteObject(pinData);
+                    }
                 }
-                else
+            }
+            catch (System.TypeInitializationException e) // Unosquare.RaspberryIO.Gpio.GpioController.Initialize throws this TypeInitializationException
+            {
+                if (!Unosquare.RaspberryIO.Computer.SystemInfo.Instance.IsRunningAsRoot)
                 {
-                    GpioPinData pinData = new GpioPinData(pinId, pinBoolValue ? SignalLevel.High : SignalLevel.Low, pin);
-                    WriteObject(pinData);
+                    throw new PlatformNotSupportedException(Resources.ErrNeedRootPrivileges, e);
                 }
             }
```

If they are running as root we should probably `throw` the error again so we don't accidentally swallow it?